src/averageAges.js

```diff
@@ -37,7 +47,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const female = people.filter(({ sex }) => sex === 'f');
+
+  const withChild = withChildren
+    ? female.filter(
+      ({ name }) => people.some(({ mother }) => mother === name))
+    : female;
+
+  const averageAge = withChild.reduce(
```

Comment:
It would be great to create separate method to calculate average age as mentioned in comments above to not repeat this code